zap/src/config.rs

```diff
@@ -230,29 +230,29 @@ impl<'src> Ty<'src> {
 				if let Some(exact) = len.exact() {
 					(exact as usize, Some(exact as usize))
 				} else {
-					let len_size = len.numty().unwrap_or(NumTy::U16).size();
+					let len_numty = len.numty().map(|(numty, ..)| numty).unwrap_or(NumTy::U16);
 
 					(
-						len.min().map(|min| min as usize).unwrap_or(0) + len_size,
-						len.max().map(|max| (max as usize) + len_size),
+						len.min().map(|min| min as usize).unwrap_or(0) + len_numty.size(),
+						len.max().map(|max| max as usize + len_numty.size()),
 					)
 				}
 			}
 
 			Self::Arr(ty, len) => {
 				let (ty_min, ty_max) = ty.size(tydecls, recursed);
 				let len_min = len.min().map(|min| min as usize).unwrap_or(0);
-				let len_size = len.numty().unwrap_or(NumTy::U16).size();
+				let len_numty = len.numty().map(|(numty, ..)| numty).unwrap_or(NumTy::U16);
 
 				if let Some(exact) = len.exact() {
 					(ty_min * (exact as usize), ty_max.map(|max| ty_max.unwrap() * max))
```

Comment:
I don't think it's your fault, but I think this is wrong as it's just doing `ty_max * ty_max`